src/components/input.js

```diff
@@ -41,21 +41,54 @@ const styles = {
   }
 }
 
-export const textInput = function(props) {
-  return h(Interactive, _.merge(
-    {
-      as: 'input',
-      className: 'focus-style',
-      style: {
-        ...styles.input,
-        width: '100%',
-        paddingLeft: '1rem', paddingRight: '1rem',
-        fontWeight: 400, fontSize: 14,
-        backgroundColor: props.disabled ? colors.grayBlue[4] : undefined
+export const textInput = props => h(Interactive,
+  _.merge({
+    as: 'input',
+    className: 'focus-style',
+    style: {
+      ...styles.input,
+      width: '100%',
+      paddingLeft: '1rem', paddingRight: '1rem',
+      fontWeight: 400, fontSize: 14,
+      backgroundColor: props.disabled ? colors.grayBlue[4] : undefined
+    }
+  },
+  props)
+)
+
+export const SearchInput = ({ onSearch = _.noop, onChange, ...props }) => {
+  const supportsSearch = 'onsearch' in document.documentElement
+
+  const searchEl = useRef()
+  const attachRef = node => {
+    if (node) {
+      const el = findDOMNode(node)
```

Comment:
Rather than explicitly getting the DOM node, we could just attach the ref directly and then set the event listener in an effect.